### PR TITLE
[yeelight] Add basic support for yeelight desklamp

### LINKED ISSUE #4452: [yeelight] Add desklamp support
I have a yellight desklamp and the binding addon does not support it, so i want to implement support for the desklamp

### PR DESCRIPTION
**Pull request by sebastianrakel**
This PR should provide basic functions of the desklamp, like the ceeling light

I have a desklamp at home and want to include it into my openhab environment

Fixes #4452

### CODE REVIEW

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    mDeviceStatus.setName(status.get(1).getAsString());
+
+                    // brightness:
+                    mDeviceStatus.setBrightness(status.get(2).getAsInt());
+
+                    // ct:
+                    mDeviceStatus.setCt(status.get(3).getAsInt());
+                }
+            }
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (sebastianrakel) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: Maybe, i only copied a other light and changed the needed lines.

I will have a look into that at the weekend

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
```
[the remark this answers is not in the document]
Comment (sebastianrakel) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: All headers in this binding or only at the new files?

[the remark this answers is not in the document]
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: The header of this file is sufficient.

[the remark this answers is not in the document]
Comment (sebastianrakel) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: Ah, sorry. Will change it at the weekend

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianrakel) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: I don't know how to catch this exception more specific, normaly this exception should not be thrown. I can remove this exception but i don't know if that is a better solution

**Comment by cweitkamp** [comment]
Thanks for the updates. Can you have a look why Travis does not like the latest changes?

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.2.0:compile (default-compile) on project org.openhab.binding.yeelight: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/services/DeviceManager.java:[193] 
[ERROR] 	} catch (IOException e) {
[ERROR] 	         ^^^^^^^^^^^
[ERROR] IOException cannot be resolved to a type
[ERROR] 1 problem (1 error)
```

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DesklampDevice.java]: Nevermind. It seems to be the way how this binding handles it. Let's do not touch it though.

**Comment by sebastianrakel** [comment]
@cweitkamp thanks. is fixed now

**Pull request merged**